tests/inc/az_ulib_ustream_forward_compliance_ut.h

```diff
@@ -0,0 +1,613 @@
+  (USTREAM_FORWARD_COMPLIANCE_EXPECTED_CONTENT_LENGTH >> 2)
+#define USTREAM_FORWARD_COMPLIANCE_LENGTH_2 \
+  (USTREAM_FORWARD_COMPLIANCE_LENGTH_1 + USTREAM_FORWARD_COMPLIANCE_LENGTH_1)
+#define USTREAM_FORWARD_COMPLIANCE_LENGTH_3 \
+  (USTREAM_FORWARD_COMPLIANCE_LENGTH_2 + USTREAM_FORWARD_COMPLIANCE_LENGTH_1)
+
+/* create local buffer with enough size to handle the full content. */
+#define USTREAM_FORWARD_COMPLIANCE_TEMP_BUFFER_LENGTH \
+  (USTREAM_FORWARD_COMPLIANCE_EXPECTED_CONTENT_LENGTH + 2)
+
+// flush callback global variables
+static uint8_t* flush_callback_buffer_check[USTREAM_FORWARD_COMPLIANCE_EXPECTED_CONTENT_LENGTH]
+    = { 0 };
+static size_t flush_callback_size_check = 0;
+static az_ulib_callback_context flush_callback_context_check;
+
+// mock calback function
+static void mock_flush_callback(
+    const uint8_t* const buffer,
+    size_t size,
+    az_ulib_callback_context flush_callback_context)
+{
+  (void)memcpy(flush_callback_buffer_check, buffer, size);
+  flush_callback_size_check = size;
+  flush_callback_context_check = flush_callback_context;
+}
+
+/*
+ * Start compliance tests:
+ */
+
+#ifndef AZ_NO_PRECONDITION_CHECKING
+/* If the provided handle is NULL, the dispose shall fail with precondition. */
+static void az_ulib_ustream_forward_dispose_compliance_null_buffer_failed(void** state)
+{
+  /// arrange
+  (void)state;
+  az_ulib_ustream_forward* ustream_forward;
+  USTREAM_FORWARD_COMPLIANCE_TARGET_FACTORY(&ustream_forward);
+
+  /// act
+  /// assert
+  AZ_ULIB_ASSERT_PRECONDITION_CHECKED((ustream_forward)->_internal.api->dispose(NULL));
+
+  /// cleanup
+  az_result result = az_ulib_ustream_forward_dispose(ustream_forward);
+  (void)result;
+}
+
+/* If the provided handle is not the implemented buffer type, the dispose shall fail with
+ * precondition. */
+static void az_ulib_ustream_forward_dispose_compliance_buffer_is_not_type_of_buffer_failed(
+    void** state)
+{
+  /// arrange
+  (void)state;
+  az_ulib_ustream_forward* ustream_forward;
+  USTREAM_FORWARD_COMPLIANCE_TARGET_FACTORY(&ustream_forward);
+
+  /// act
+  /// assert
+  AZ_ULIB_ASSERT_PRECONDITION_CHECKED(
+      ustream_forward->_internal.api->dispose(ustream_forward_mock_create()));
+
+  /// cleanup
+  az_result result = az_ulib_ustream_forward_dispose(ustream_forward);
+  (void)result;
+}
+
+/* If the provided handle is NULL, the get_size shall fail with precondition. */
+static void az_ulib_ustream_forward_get_size_compliance_null_buffer_failed(void** state)
+{
+  /// arrange
+  (void)state;
+  az_ulib_ustream_forward* ustream_forward;
+  USTREAM_FORWARD_COMPLIANCE_TARGET_FACTORY(&ustream_forward);
+
+  /// act
+  /// assert
+  AZ_ULIB_ASSERT_PRECONDITION_CHECKED(ustream_forward->_internal.api->get_size(NULL));
+
+  /// cleanup
+  az_result result = az_ulib_ustream_forward_dispose(ustream_forward);
+  (void)result;
+}
+
+static void az_ulib_ustream_forward_get_size_compliance_buffer_is_not_type_of_buffer_failed(
+    void** state)
```

Comment:
>state [](http://example.com/codeflow?start=11&length=5)

Will the comment /* If the provided handle is NULL, the flush shall fail with precondition. */ also be applicable to this function?